File: apps/web/src/types/provider.ts
export interface ProviderModel {
  id: string;
  provider_id: string;
  model_id: string;
  display_name: string;
  is_manual: boolean;
  enabled: boolean;
}

export interface Provider {
  id: string;
  name: string;
  provider_type: string;
  base_url?: string;
  has_api_key: boolean;
  enabled: boolean;
  is_builtin: boolean;
  created_at: string;
  updated_at: string;
  models: ProviderModel[];
}

export interface ProviderCreate {
  name: string;
  provider_type: string;
  base_url?: string;
  api_key?: string;
  extra_headers_json?: string;
}

export interface ProviderUpdate {
  name?: string;
  base_url?: string;
  api_key?: string;
  clear_api_key?: boolean;
}

export interface ModelUpdate {
  display_name?: string;
  enabled?: boolean;
  mode?: string;
}

File: apps/api/app/schemas/provider.py
from datetime import datetime
from pydantic import BaseModel, Field


class ProviderModelSchema(BaseModel):
    model_config = {"from_attributes": True}

    id: str
    provider_id: str
    model_id: str
    display_name: str
    is_manual: bool
    enabled: bool


class ProviderSchema(BaseModel):
    model_config = {"from_attributes": True}

    id: str
    name: str
    provider_type: str
    base_url: str
    has_api_key: bool
    extra_headers_json: str
    enabled: bool
    is_builtin: bool
    created_at: datetime
    updated_at: datetime
    models: list[ProviderModelSchema] = []


class ProviderCreate(BaseModel):
    name: str = Field(min_length=1, max_length=200)
    provider_type: str = "openai_compatible"
    base_url: str = ""
    api_key: str = ""
    extra_headers_json: str = "{}"
    enabled: bool = True


class ProviderUpdate(BaseModel):
    name: str | None = Field(default=None, max_length=200)
    provider_type: str | None = None
    base_url: str | None = None
    api_key: str | None = None
    clear_api_key: bool = False
    extra_headers_json: str | None = None
    enabled: bool | None = None


class ProviderModelCreate(BaseModel):
    model_id: str = Field(min_length=1, max_length=200)
    display_name: str = ""


class ProviderModelUpdate(BaseModel):
    display_name: str | None = None
    enabled: bool | None = None


class TestConnectionResponse(BaseModel):
    success: bool
    message: str
